src/s3_plugin.js

```diff
@@ -84,29 +88,27 @@ module.exports = class S3Plugin {
                              compiler.options.output.context ||
                              '.'
 
-    compiler.plugin('after-emit', (compilation, cb) => {
+    compiler.hooks.afterEmit.tapPromise(packageJson.name, async(compilation) => {
```

Comment:
Marking these functions async is unnecessary no?

Reply:
It makes sure that if they will throw, it will turn into Promise.reject